lms/djangoapps/instructor/cybersource_enrollment_report.py

```diff
@@ -0,0 +1,172 @@
+"""
+Defines concrete class for cybersource  Enrollment Report.
+
+"""
+from courseware.access import has_access
+import collections
+from django.utils.translation import ugettext as _
+from courseware.courses import get_course_by_id
+from instructor.enrollment_report import BaseEnrollmentReportProvider
+from shoppingcart.models import RegistrationCodeRedemption, PaidCourseRegistration, CouponRedemption, OrderItem, \
+    InvoiceTransaction
+from student.models import CourseEnrollment
+
+
+class CyberSourceEnrollmentReportProvider(BaseEnrollmentReportProvider):
+    """
+    The concrete class for all CyberSource Enrollment Reports.
+    """
+
+    def get_enrollment_info(self, user, course_id):
+        """
+        Returns the User Enrollment information.
+        """
+        course = get_course_by_id(course_id, depth=0)
+        is_course_staff = has_access(user, 'staff', course)
+
+        # check the user enrollment role
+        if user.is_staff:
+            enrollment_role = _('Edx Staff')
+        elif is_course_staff:
+            enrollment_role = _('Course Staff')
+        else:
+            enrollment_role = _('Student')
+
+        course_enrollment = CourseEnrollment.get_enrollment(user=user, course_key=course_id)
+
+        if is_course_staff:
+            enrollment_source = _('Staff')
+        else:
+            # get the registration_code_redemption object if exists
+            registration_code_redemption = RegistrationCodeRedemption.registration_code_used_for_enrollment(
+                course_enrollment)
+            # get the paid_course registration item if exists
+            paid_course_reg_item = PaidCourseRegistration.get_course_item_for_user_enrollment(
+                user=user,
+                course_id=course_id,
+                course_enrollment=course_enrollment
+            )
+
+            # from where the user get here
+            if registration_code_redemption is not None:
+                enrollment_source = _('Used Registration Code')
+            elif paid_course_reg_item is not None:
+                enrollment_source = _('Credit Card - Individual')
+            else:
+                enrollment_source = _('Manually Enrolled')
+
+        enrollment_date = getattr(course_enrollment, 'created').strftime("%B %d, %Y")
```

Comment:
Why the use of `getattr`? Both `created` and `is_active` are valid model fields.